cg/cg_iterative_solvers.jl

```diff
@@ -1,8 +1,7 @@
 using IterativeSolvers
 
 function cg_iterative_solvers_kernel(x, A, b, l)
-    IterativeSolvers.cg!(x, A, b; abstol = zero(eltype(A)), reltol = zero(eltype(A)), maxiter = l, log = false)
-    x
+    IterativeSolvers.cg(A, b; abstol = zero(eltype(A)), reltol = zero(eltype(A)), maxiter = l, log = false)
```

Comment:
Just to be extra sure, could we print out how many iterations IterativeSolvers is performing?

Reply:
So it doesn't seem like there is a way to get the number of iterations when using the IterativeSolvers library because it does not return it, but I had copied cg.jl from IterativeSolvers earlier and manually inserted a print statement in the code to confirm. IterativeSolvers uses `iteration ≥ it.maxiter || converged(it)` to determine when to stop iterating where `converged()` checks if the residual is less than the tolerance.

With `maxiter` being set to only 20, we don't get to a point where the residual is less than the tolerance, but I can add a similar residual < tolerance check in the Finch implementation to make it comparable just in case.

Reply:
Actually never mind, I think we can get the number of iterations...